src/pages/Admin/Tools/Tools.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import firebase from "firebase/app";
+
+import * as scripts from "./scripts";
+import { SelfServeScript } from "./types";
+
+export const Tools: React.FC = () => {
+  const [chosenScript, setChosenScript] = useState<SelfServeScript>();
+
+  return (
+    <div>
+      {chosenScript ? (
+        <Script script={chosenScript} />
+      ) : (
+        Object.values(scripts).map((script) => (
+          <button
+            onClick={() => setChosenScript(script)}
+            key={`${script.name}-${script.functionLocation}`}
+          >
+            {script.name}
+          </button>
+        ))
+      )}
+    </div>
+  );
+};
+
+export const Script: React.FC<{ script: SelfServeScript }> = ({ script }) => {
+  const { register, handleSubmit } = useForm();
+
+  const [links, setLinks] = useState<{ [key: string]: string }>();
+
+  const onSubmit = async (data: Object) => {
+    console.log({ data });
+    const response = await firebase
+      .functions()
+      .httpsCallable(script.functionLocation)(data);
+
+    console.log(response.data);
+
+    setLinks(response.data);
+  };
+
+  if (links) {
+    return (
+      <div>
+        {Object.entries(links).map(([name, link]) => (
+          <p key={`${name}:${link}`}>
+            {name} &nbsp;
+            <a href={link}>Download</a>
+          </p>
+        ))}
+      </div>
+    );
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Can benefit from `useForm()` not necessarily required for this PR

Reply:
Not sure what do you mean by this

Reply:
A more elaborate form (example, like in `<WorldGeneralForm/>`) that has `useForm` and `useAsyncFn` hooks to deal more robustly with the input. Not necessarily in scope for this PR, but maybe for a next one